lib/datadog/appsec/contrib/rack/gateway/request.rb

```diff
@@ -36,7 +36,7 @@ def query
             # Rack < 2.0 does not have :each_header
             # TODO: We need access to Rack here. We must make sure we are able to load AppSec without Rack,
             # TODO: while still ensure correctness in ths code path.
-            if defined?(::Rack) && ::Rack::Request.instance_methods.include?(:each_header)
```

Comment:
It would be very helpful for us to know more about the setup you have running to make sure we apply the correct fix to this situation. 


Are you enabling the appsec Rack, Sinatra or Rails integration?

Reply:
I think this check here should be changed to check for `::Rack::Request`:
https://github.com/DataDog/dd-trace-rb/blob/25686d40586c992e8950113b8e80483633341dc2/lib/datadog/appsec/contrib/rack/integration.rb#L25

Because the application has not loaded any instrumentable Rack classes, we should not consider that Rack is loaded.

Reply:
Thanks for investigating this.

The system we're having trouble with is an internal integration service. It doesn't expose an external endpoint (HTTP or otherwise). Rather, it has a number of scheduled and asynchronous tasks that it runs. It uses the [sidekiq](https://github.com/sidekiq/sidekiq) and [clockwork](https://github.com/Rykian/clockwork) gems to support this functionality.

Rack is pulled in via a transitive dependency. As is the `rack/utils` require.

Reply:
In the config, I see a number of `config.tracing.instrument` calls: active_support, aws, http, redis, sidekiq, and faraday.